pkg/sqlite/query.go

```diff
@@ -752,3 +780,103 @@ func (s *SQLQuerier) GetCurrentCSVNameForChannel(ctx context.Context, pkgName, c
 	}
 	return "", nil
 }
+
+func (s *SQLQuerier) ListBundles(ctx context.Context) (bundles []*api.Bundle, err error) {
+	query := `SELECT DISTINCT channel_entry.entry_id, operatorbundle.bundle, operatorbundle.bundlepath, MIN(channel_entry.depth), channel_entry.operatorbundle_name, channel_entry.package_name, channel_entry.channel_name, channel_entry.replaces, operatorbundle.version, operatorbundle.skiprange
+	FROM channel_entry
+	INNER JOIN operatorbundle ON operatorbundle.name = channel_entry.operatorbundle_name
+	INNER JOIN package ON package.name = channel_entry.package_name
+	GROUP BY channel_entry.package_name, channel_entry.channel_name`
+
+	rows, err := s.db.QueryContext(ctx, query)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	bundles = []*api.Bundle{}
+	for rows.Next() {
+		var entryID sql.NullInt64
+		var bundle sql.NullString
+		var bundlePath sql.NullString
+		var minDepth sql.NullInt64
+		var bundleName sql.NullString
+		var pkgName sql.NullString
+		var channelName sql.NullString
+		var replaces sql.NullString
+		var version sql.NullString
+		var skipRange sql.NullString
+		if err := rows.Scan(&entryID, &bundle, &bundlePath, &minDepth, &bundleName, &pkgName, &channelName, &replaces, &version, &skipRange); err != nil {
+			return nil, err
+		}
+
+		out := &api.Bundle{}
+		if bundle.Valid && bundle.String != "" {
+			out, err = registry.BundleStringToAPIBundle(bundle.String)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		out.CsvName = bundleName.String
+		out.PackageName = pkgName.String
+		out.ChannelName = channelName.String
+		out.BundlePath = bundlePath.String
+		out.Version = version.String
+		out.SkipRange = skipRange.String
+
+		provided, required, err := s.GetApisForEntry(ctx, entryID.Int64)
+		if err != nil {
+			return nil, err
+		}
+		out.ProvidedApis = provided
+		out.RequiredApis = required
+
+		dependencies, err := s.GetDependenciesForBundle(ctx, bundleName.String, version.String, bundlePath.String)
```

Comment:
Do we need to run a separate query here for each row? Can't we just do a join up front to get this table data as well?

Reply:
We have `one to many` problem here with operatorbundle to multiple dependencies similar to the API table. The trade off will be a complex nested loop to read multiple rows of the same operatorbundle but with each dependency.

Reply:
From a performance perspective the cost is definitely going to be higher doing that many queries. Can't we just have the aggregation be a row per dependency and iterate over them one at a time? We have to do that bit anyway